File: Programmers/StackQueue/printer.py
def solution(priorities, location):
    tmp = []
    order = []
    for i in range(len(priorities)):
        tmp.append(i)
    while(len(priorities) != 0):
        if(len(priorities) == 1):
            order.append(tmp[0])
            break
        maximum = max(priorities[1::])
        if priorities[0] >= maximum:
            order.append(tmp[0])
            priorities = priorities[1::]
            tmp = tmp[1::]
        else:
            priorities.append(priorities[0])
            tmp.append(tmp[0])
            priorities = priorities[1::]
            tmp = tmp[1::]
    return order.index(location) + 1

print(solution([2, 1, 3, 2], 2))
print(solution([1, 1, 9, 1, 1, 1], 0))

File: Programmers/StackQueue/printer.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> priorities, int location)
{
    vector<int> order;
    vector<int> temp;
    for (int i = 0; i < priorities.size(); i++)
    {
        temp.push_back(i);
    }
    while (!priorities.empty())
    {
        if (priorities.size() == 1)
        {
            order.push_back(temp.front());
            temp.pop_back();
            priorities.pop_back();
            break;
        }
        int max = *max_element(priorities.begin() + 1, priorities.end());
        if (priorities.front() >= max)
        {
            order.push_back(temp.front());
            priorities = vector<int>(priorities.begin() + 1, priorities.end());
            temp = vector<int>(temp.begin() + 1, temp.end());
        }
        else
        {
            priorities.push_back(priorities.front());
            temp.push_back(temp.front());
            priorities = vector<int>(priorities.begin() + 1, priorities.end());
            temp = vector<int>(temp.begin() + 1, temp.end());
        }
    }
    for (const auto &num : order)
    {
        cout << num << " ";
    }
    cout << endl;
    return find(order.begin(), order.end(), location) - order.begin() + 1;
}

int main()
{
    vector<int> priorities = {1, 1, 2, 3, 2, 1};
    cout << solution(priorities, 0) << endl;
    cin.get();
}